subprojects/core/src/main/java/org/gradle/initialization/SettingsFactory.java

```diff
@@ -59,4 +61,19 @@ public SettingsInternal createSettings(GradleInternal gradle, File settingsDir,
         ((ExtensibleDynamicObject) dynamicObject).addProperties(gradleProperties);
         return settings;
     }
+
+
+    private ClassLoaderScope cloneScope(ClassLoaderScope originScope) {
+        if (originScope instanceof DefaultClassLoaderScope) {
+            DefaultClassLoaderScope origin = (DefaultClassLoaderScope) originScope;
+            DefaultClassLoaderScope clonedScope = new DeprecatingClassLoaderScope(origin.getId(), origin.getParent(), origin.getClassLoaderCache());
```

Comment:
Maybe we just add a `deprecated()` to ClassLoaderScope to return a ClassLoaderScope that knows it's deprecated and would create the appropriate deprecation-producing classloader as necessary?

Reply:
thats what I went with. updated the code

Reply:
Maybe I'm misunderstanding, but I think what Sterling was recommending was that you would be able to just call `buildRootLoaderScope.deprecated()` to get a deprecated `ClassLoaderScope` based on `buildRootLoaderScope` instead of having to construct one.  If a scope object was already deprecated, it would just return itself.

Or maybe you just didn't push your change?

Reply:
You're a wise man. Indeed I havn't pushed my changes yet. Did that now. Will ping Adam to add his reservations here